tests/test_lib_tokens_webauthn.py

```diff
@@ -0,0 +1,365 @@
+        self.init_params['nonce'] = webauthn_b64_decode(REGISTRATION_CHALLENGE)
+        self.token.get_init_detail(self.init_params, self.user)
+        self.token.update({
+            'type': 'webauthn',
+            'serial': self.token.token.serial,
+            'regdata': REGISTRATION_RESPONSE_TMPL['attObj'],
+            'clientdata': REGISTRATION_RESPONSE_TMPL['clientData'],
+            WEBAUTHNACTION.RELYING_PARTY_ID: RP_ID,
+            WEBAUTHNACTION.AUTHENTICATOR_ATTESTATION_LEVEL: ATTESTATION_LEVEL.NONE,
+            'HTTP_ORIGIN': ORIGIN
+        })
+        web_authn_registration_response = self.token.get_init_detail().get("webAuthnRegisterResponse")
+
+        self.assertTrue(web_authn_registration_response
+                        .get("subject")
+                        .startswith("Yubico U2F EE Serial"))
+        self.assertTrue(self
+                        .token
+                        .get_tokeninfo(WEBAUTHNINFO.ATTESTATION_ISSUER)
+                        .startswith("CN=Yubico U2F Root CA Serial"))
+        self.assertTrue(self
+                        .token
+                        .get_tokeninfo(WEBAUTHNINFO.ATTESTATION_SUBJECT)
+                        .startswith("CN=Yubico U2F EE Serial"))
+        self.assertEqual(CRED_ID, self.token.decrypt_otpkey())
+        self.assertEqual(PUB_KEY, self.token.get_tokeninfo(WEBAUTHNINFO.PUB_KEY))
+
+    def test_04_authentication(self):
+        web_authn_sign_request = self._create_challenge().get("webAuthnSignRequest")
+
+        self.assertEqual(RP_ID, web_authn_sign_request.get('rpId'))
+        self.assertEqual(1, len(web_authn_sign_request.get('allowCredentials') or []))
+        self.assertEqual('public-key', web_authn_sign_request.get('allowCredentials')[0].get('type'))
+        self.assertEqual(CRED_ID, web_authn_sign_request.get('allowCredentials')[0].get('id'))
+        self.assertEqual(ALLOWED_TRANSPORTS, web_authn_sign_request.get('allowCredentials')[0].get('transports'))
+
+    def test_05_authorization(self):
+        self.challenge_options['nonce'] = webauthn_b64_decode(ASSERTION_CHALLENGE)
+        self._create_challenge()
+        sign_count = self.token.check_otp(otpval=None,
+                                          options={
+                                              "credentialid": CRED_ID,
+                                              "authenticatordata": ASSERTION_RESPONSE_TMPL['authData'],
+                                              "clientdata": ASSERTION_RESPONSE_TMPL['clientData'],
+                                              "signaturedata": ASSERTION_RESPONSE_TMPL['signature'],
+                                              "user": self.user,
+                                              "challenge": hexlify_and_unicode(self.challenge_options['nonce']),
+                                              "HTTP_ORIGIN": ORIGIN,
+                                          })
+        self.assertTrue(sign_count > 0)
+
+#
```

Comment:
THe change of the copyright in the file seems awkward. Can we put this into a 2nd file?